### PR TITLE
Code has references to Sun and Solaris

### PR DESCRIPTION
**Pull request by jrw972**
Problem
-------

Sun and Solaris are no longer supported.

Solution
--------

Remove references to Sun and Solaris.

### CODE REVIEW

File: dds/DCPS/SafetyProfileSequence.h [hunk 1]
```diff
@@ -311,10 +311,6 @@ namespace SafetyProfile {
     // max_size() inherited from AllocPolicy
     bool empty() const { return !length_; }
 
-#ifndef __SUNPRO_CC
```
Comment (mitza-oci) [inline dds/DCPS/SafetyProfileSequence.h:314]: This is `ifndef`, not `ifdef`

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: dds/InfoRepo/FederatorManagerImpl.h [hunk 2]
```diff
@@ -137,8 +137,6 @@ class OpenDDS_Federator_Export ManagerImpl
   //
   // UpdateProcessor<> methods.
   //
-  // "using" directive to fix "Hides the virtual function in virtual base" warning on
-  // SunOS compiler.
   using UpdateProcessor<OwnerUpdate>::processUpdateQos2;
```
Comment (ocielliottc) [inline dds/InfoRepo/FederatorManagerImpl.h:140]: Should these be removed if they were put in place for the Sun compiler?

**Review by mitza-oci — APPROVED** [review]
Should be ready to go when CI passes.  See https://github.com/objectcomputing/OpenDDS/actions/runs/3832882423/jobs/6523708896

File: dds/InfoRepo/FederatorManagerImpl.h (hunk 2 above)
Comment (iguessthislldo) [reply dds/InfoRepo/FederatorManagerImpl.h]: Looks like Apple clang complains about it as well.

**Pull request merged**